### ISSUE TITLE
Limited Mystery Box

### ISSUE THREAD

**Issue by TheTrickyDonut** [Zet0rz/nzombies#404]
Tested with CW 2.0 weapons, the mystery box can only give as much weapons as you added before saying something like "No weapons available". Forcing to press E will only yield a teddy bear. To put it into perspective, when I only include a pistol and a shotgun to the weapons that will be rolled, then the box in turn will only let you buy two times before saying that message of weapons no longer being available. Simply put - Weapons added is proportional to how many times you can buy from the box. At first, I thought this only occured because I only had one box in the map, but it still persisted when I added more. Any help to alleviate this limitation to purchase weapons will be greatly appreciated.

**Comment by minin43** [comment]
What

**Comment by globalcringe** [comment]
Can confirm, just tested. Reproduction steps:
1. Put, say, five guns in box.
2. Buy the box five times.
3. Even if you got the same gun five times, it'll always say "No available weapons left!" after buying it a sixth time.
Not really noticeable on boxes with loads of guns, but on larger servers this can be a problem.
Getting a GAME OVER! does not fix this, only a config reload or map change will.

**Comment by Zet0rz** [comment]
Very interesting, but highly likely to be caused by conflicting addons. There has been no update to the random box system since last it was confirmed working. It's supposed to only blacklist weapons you are currently carrying, so switching weapons out should add that weapon to the box again. Will try to reproduce and see if it is a bug in the gamemode.

**Comment by TheTrickyDonut** [comment]
Tried disabling all addons except the content pack, CW 2.0, and FAS 2.0. Unfortunately, it still persists. I tried making a workaround by adding a LOT of other weapons and setting their box spawn chances to (0), and setting all my CW 2.0 to default chances (10) and see if it will work. In a way, it did work. It increases the times I can purchase, but after the box goes through all the CW 2.0 weapons, it forces the other weapons that I set to (0) chances to spawn. Ideally, I would only limit the box to CW 2.0 (Because I love this weapon pack), but then again, I'm only limited to around 20+ purchases (Since that's the only number of weapons the pack has). I would appreciate a fix for this since I really enjoy your gamemode.
